machineconfiguration/v1/types_machineosconfig.go

```diff
@@ -0,0 +1,243 @@
+	// +kubebuilder:validation:XValidation:rule=`(self.split('@')[0].matches('^([a-zA-Z0-9-]+\\.)+[a-zA-Z0-9-]+(:[0-9]{2,5})?/([a-zA-Z0-9-_]{0,61}/)?[a-zA-Z0-9-_.]*?$'))`,message="the OCI Image name should follow the host[:port][/namespace]/name format, resembling a valid URL without the scheme"
+	// +optional
+	BaseOSExtensionsImagePullspec string `json:"baseOSExtensionsImagePullspec,omitempty"`
+	// baseOSImagePullspec is the base OSImage we use to build our custom image.
+	// the MachineOSConfig object will use the in cluster image registry configuration.
+	// if you wish to use a mirror or any other settings specific to registries.conf, please specify those in the cluster wide registries.conf.
+	// The format of the image pullspec is:
+	// host[:port][/namespace]/name@sha256:<digest>
+	// The digest must be 64 characters long, and consist only of lowercase hexadecimal characters, a-f and 0-9.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=447
+	// +kubebuilder:validation:XValidation:rule=`(self.split('@').size() == 2 && self.split('@')[1].matches('^sha256:[a-f0-9]{64}$'))`,message="the OCI Image reference must end with a valid '@sha256:<digest>' suffix, where '<digest>' is 64 characters long"
+	// +kubebuilder:validation:XValidation:rule=`(self.split('@')[0].matches('^([a-zA-Z0-9-]+\\.)+[a-zA-Z0-9-]+(:[0-9]{2,5})?/([a-zA-Z0-9-_]{0,61}/)?[a-zA-Z0-9-_.]*?$'))`,message="the OCI Image name should follow the host[:port][/namespace]/name format, resembling a valid URL without the scheme"
+	// +optional
+	BaseOSImagePullspec string `json:"baseOSImagePullspec,omitempty"`
+	// baseImagePullSecret is the secret used to pull the base image.
+	// must live in the openshift-machine-config-operator namespace if provided.
+	// defaults to using the cluster-wide pull secret if not specified.
+	// +optional
+	BaseImagePullSecret *ImageSecretObjectReference `json:"baseImagePullSecret,omitempty"`
+	// machineOSImageBuilder describes which image builder will be used in each build triggered by this MachineOSConfig
+	// Currently supported type(s): JobImageBuilder
+	// +kubebuilder:validation:Required
+	ImageBuilder *MachineOSImageBuilder `json:"imageBuilder"`
+	// renderedImagePushSecret is the secret used to connect to a user registry.
+	// the final image push and pull secrets should be separate for security concerns. If the final image push secret is somehow exfiltrated,
+	// that gives someone the power to push images to the image repository. By comparison, if the final image pull secret gets exfiltrated,
+	// that only gives someone to pull images from the image repository. It's basically the principle of least permissions.
+	// this push secret will be used only by the MachineConfigController pod to push the image to the final destination. Not all nodes will need to push this image, most of them
+	// will only need to pull the image in order to use it.
```

Comment:
I wonder if we can make this more direct, but still convey the same message, what about?
```suggestion
	// The final image push and pull secrets should be separate and assume the principal of least privilege.
	// The push secret with write privilege is only required to be present on the node hosting the MachineConfigController pod.
	// The pull secret with read only privileges is required on all nodes.
	// By separating the two secrets, the risk of write credentials becoming compromised is reduced.
```